src/Agoda.Analyzers.Test/AgodaCustom/AG0010UnitTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Agoda.Analyzers.AgodaCustom;
+using Agoda.Analyzers.Test.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Framework;
+
+namespace Agoda.Analyzers.Test.AgodaCustom
+{
+    internal class AG0010UnitTests : DiagnosticVerifier
+    {
+        private const string TEST_FIXTURE_ATTRIBUTE = "TestFixture";
+        private const string TEST_INHERITANCE_CLASSNAME = "BaseTest";
+
+        [Test]
+        public async Task AG0010_WhenNoTestFixtureAttribute_ShouldntShowAnyWarning()
+        {
+            var code = ClassBuilder.New()
+                .WithNamespace()
+                .WithClass()
+                .Build();
+
+            await TestWarnings(code);
+        }
+
+        [Test]
+        public async Task AG0010_WhenPutTestFixtureAttribute_ButNoInheritance_ShouldntShowWarning()
+        {
+            var code = ClassBuilder.New()
+                .WithNamespace()
+                .WithClass(className: "test", attribute: TEST_FIXTURE_ATTRIBUTE)
+                .WithAttributeClass(TEST_FIXTURE_ATTRIBUTE)
+                .Build();
+
+            await TestWarnings(code);
+        }
+
+        [Test]
+        public async Task AG0010_WhenPutTestFixtureAttribute_ButHasInheritance_ShouldShowWarning()
```

Comment:
This is close, but not quite there yet. The `TestFixture` attribute is optional in NUnit. Instead, we need to check if one of of the methods in the class is a test case. This might help you: https://github.com/agoda-com/AgodaAnalyzers/blob/master/src/Agoda.Analyzers/Helpers/MethodHelper.cs

Reply:
Already changed